vowpalwabbit/core/src/parse_args.cc

```diff
@@ -419,27 +421,58 @@ input_options parse_source(VW::workspace& all, options_i& options)
   // Check if the options provider has any positional args. Only really makes sense for command line, others just return
   // an empty list.
   const auto positional_tokens = options.get_positional_tokens();
-  if (positional_tokens.size() == 1) { all.data_filename = positional_tokens[0]; }
-  else if (positional_tokens.size() > 1)
-  {
-    all.logger.err_warn(
-        "Multiple data files passed as positional parameters, only the first one will be "
-        "read and the rest will be ignored.");
-  }
 
-  if (parsed_options.daemon || options.was_supplied("pid_file") || (options.was_supplied("port") && !all.active))
+  if (!positional_tokens.empty() && !all.data_filenames.empty())
+  { THROW("You cannot mix --data/-d argument with positional arguments. Pick one and use consistently."); }
```

Comment:
added this throw to allow users to pick either all --data or positional, would this be helpful as a previous step to deprecating positional?